### PR TITLE
[sw/silicon_creator] Update keymgr functest

### PR DESCRIPTION
**Pull request by moidx**
Update the keymgr functest to use software initiated reset instead of
low power entry/exit. Also update flash and OTP configuration to use
available device testutils.

### CODE REVIEW

File: sw/device/silicon_creator/lib/drivers/keymgr_functest.c [hunk 1]
```diff
@@ -301,37 +203,31 @@ rom_error_t keymgr_rom_ext_test(void) {
...
-
-  } else {
+    dif_otp_ctrl_t otp;
+    CHECK_DIF_OK(dif_otp_ctrl_init(
+        mmio_region_from_addr(TOP_EARLGREY_OTP_CTRL_CORE_BASE_ADDR), &otp));
+    otp_ctrl_testutils_lock_partition(&otp, kDifOtpCtrlPartitionSecret2, 0);
+
+    // Issue and wait for reset.
+    CHECK_DIF_OK(dif_rstmgr_reset_info_clear(&rstmgr));
+    CHECK_DIF_OK(dif_rstmgr_software_device_reset(&rstmgr));
```
Comment (tjaychen) [inline sw/device/silicon_creator/lib/drivers/keymgr_functest.c:228]: nice!

**Comment by tjaychen** [comment]
> 

hmmm.... my best guess would be that somehow the flash / otp values are not retaining in fpga somehow when we issue a soft reset.  Are you saying if you use the low power entry/exit (changing nothing else), this works on fpga?

**Comment by moidx** [comment]
Flash  info pages seem to retain data after reset. OTP lock and digest also seem to work. I now suspect the issue is with the FPGA OTP initial values. I am going to try to load the data into OTP as part of the test to double check.

**Comment by tjaychen** [comment]
> Flash info pages seem to retain data after reset. OTP lock and digest also seem to work. I now suspect the issue is with the FPGA OTP initial values. I am going to try to load the data into OTP as part of the test to double check.

yeah unlike verilator these values don't get loaded.
But all the values you're interested in pumping into `keymgr` i would think get loaded during the test right?

**Review by alphan — APPROVED** [review]
Thanks @moidx! It would be great if we could use silicon_creator drivers wherever they provide the required functionality to increase coverage, e.g. flash_ctrl, rstmgr, etc.

File: sw/device/silicon_creator/lib/drivers/keymgr_functest.c [hunk 2]
```diff
@@ -7,11 +7,10 @@
 
 #include "sw/device/lib/arch/device.h"
 #include "sw/device/lib/base/macros.h"
-#include "sw/device/lib/dif/dif_aon_timer.h"
 #include "sw/device/lib/dif/dif_flash_ctrl.h"
 #include "sw/device/lib/dif/dif_kmac.h"
 #include "sw/device/lib/dif/dif_otp_ctrl.h"
-#include "sw/device/lib/dif/dif_pwrmgr.h"
+#include "sw/device/lib/dif/dif_rstmgr.h"
```
Comment (alphan) [inline sw/device/silicon_creator/lib/drivers/keymgr_functest.c:13]: Should also include `sw/device/lib/testing/otp_ctrl_testutils.h`

File: sw/device/silicon_creator/lib/drivers/keymgr_functest.c [hunk 3]
```diff
@@ -235,22 +150,24 @@ static void init_kmac_for_keymgr(void) {
...
+  CHECK_DIF_OK(dif_otp_ctrl_is_digest_computed(otp, kDifOtpCtrlPartitionSecret2,
+                                               &is_computed));
+  if (is_computed) {
+    uint64_t digest;
+    CHECK_DIF_OK(
+        dif_otp_ctrl_get_digest(otp, kDifOtpCtrlPartitionSecret2, &digest));
+    LOG_INFO("OTP partition locked.");
+    return;
+  }
+  otp_ctrl_testutils_lock_partition(&otp, kDifOtpCtrlPartitionSecret2, 0);
```
Comment (alphan) [inline sw/device/silicon_creator/lib/drivers/keymgr_functest.c:170]: First argument should be a pointer not a pointer-to-pointer:
```suggestion
  otp_ctrl_testutils_lock_partition(otp, kDifOtpCtrlPartitionSecret2, 0);
```

**Review by alphan — COMMENTED** [review]
@moidx I think there was a small typo in `check_lock_otp_partition()` ptal

File: sw/device/silicon_creator/lib/drivers/keymgr_functest.c (hunk 2 above)
Comment (moidx) [reply sw/device/silicon_creator/lib/drivers/keymgr_functest.c:13]: Done.

File: sw/device/silicon_creator/lib/drivers/keymgr_functest.c (hunk 3 above)
Comment (moidx) [reply sw/device/silicon_creator/lib/drivers/keymgr_functest.c]: oops! done

**Pull request merged**